core/sodasql/cli/ingest.py

```diff
@@ -232,13 +342,35 @@ def ingest(
     if not soda_server_client.api_key_id or not soda_server_client.api_key_secret:
         raise ValueError("Missing Soda cloud api key id and/or secret.")
 
-    if tool == 'dbt':
-        dbt_manifest, dbt_run_results = resolve_artifacts_paths(
-            dbt_artifacts=dbt_artifacts,
-            dbt_manifest=dbt_manifest,
-            dbt_run_results=dbt_run_results
-        )
-        test_results_iterator = map_dbt_test_results_iterator(dbt_manifest, dbt_run_results)
+    if tool == "dbt":
+        if (
+            dbt_artifacts is not None
+            or dbt_manifest is not None
+            or dbt_run_results is not None
+        ):
+            manifest, run_results = load_dbt_artifacts(
+                dbt_artifacts,
+                dbt_manifest,
+                dbt_run_results,
+            )
+        else:
+            if dbt_cloud_api_token is None:
+                raise ValueError(
+                    f"Expecting an DBT cloud api token. Currently, dbt_cloud_api_token={dbt_cloud_api_token}."
+                )
+            elif dbt_cloud_account_id is None:
+                raise ValueError(
+                    f"Expecting an DBT cloud api token. Currently, dbt_cloud_account_id={dbt_cloud_account_id}."
+                )
+            elif dbt_cloud_run_id is None:
+                raise ValueError(
```

Comment:
Perhaps a better way is to collect the errors and then print them instead of failing early:

(untested code):

```
errors = [f"Invalid value provided for {x=}" for x in
          [dbt_cloud_api_token, dbt_cloud_account_id, dbt_cloud_run_id] if x is None]
if errors:
    raise ValueError("\n".join(errors))
else:
    manifest, run_results = download_dbt_artifacts_from_cloud(
        dbt_cloud_api_token, dbt_cloud_account_id, dbt_cloud_run_id
    )
```

Reply:
hmm, yes, I would actually prefer to push this down to the click level, something [like this](https://stackoverflow.com/questions/55584012/python-click-dependent-options-on-another-option) maybe

Reply:
While click dealing with parameter errors is good, it won't help when we expose API for dbt ingestion. So I think it is better to change at the code at this level.

Reply:
ok, I'll change it